### ISSUE TITLE
Bug in MqttTcpServerAdapter

### ISSUE THREAD

**Issue by erikkj** [chkr1011/MQTTnet#741]
### Describe the bug
After upgrading to 3.0.6 I am no longer able to start the hosted MQTT server with encrypted endpoint.

### Which project is your bug related to?
- Server

### To Reproduce
1. Using this version of MQTTnet '3.0.6'.
2. Build server options with MqttServerOptionsBuilder
3. WithEncryptionCertificate(certificate)
4. Server is unable to start due to null reference exception in MqttTcpServerAdapter line 51

### Code example
Can be fixed with this code (line 51 in MqttTcpServerAdapter)
 
```csharp
// Original
var tlsCertificate = new X509Certificate2(options.TlsEndpointOptions.Certificate, options.TlsEndpointOptions.CertificateCredentials.Password);
// Modified
var tlsCertificate = new X509Certificate2(options.TlsEndpointOptions.Certificate, options.TlsEndpointOptions.CertificateCredentials?.Password);
```

**Comment by SeppPenner** [comment]
@erikkj Are you using a certificate without password? Or when exactly does this exception occur? This seems to be an with `CertificateCredentials` being null by default and trying to access it in `MqttTcpServerAdapter`.

In my opinion, this should be fixed with something like this:

```csharp
if (options.TlsEndpointOptions.CertificateCredentials == null)
{
    throw new ArgumentException("The certificate credentials are not set.");
}
```

**Comment by erikkj** [comment]
@SeppPenner
I export the certificate like this: ```new X509Certificate2(_settings.CertificatePath, _settings.CertificatePassword, X509KeyStorageFlags.Exportable).Export(X509ContentType.Pfx)```

The final certificate I send to `WithEncryptionCertificate` is without password.

The code ran smooth before the upgrade without using credentials.

**Comment by SeppPenner** [comment]
@erikkj Okay, I didn't use this without password yet. If this was possible before, you're right.

**Comment by chkr1011** [comment]
@SeppPenner @erikkj Please try again with 3.0.7-rc1. It should work again.

**Comment by danehnert** [comment]
I had the same problem. With 3.0.7-rc1 it is now working again for me.

**Comment by erikkj** [comment]
I can confirm that the problem is fixed in 3.0.7-rc1. Thank you!

**Comment by SeppPenner** [comment]
Good to know ;)
